doc/src/Developer_write_pair.rst

```diff
@@ -0,0 +1,1196 @@
+type specific data, the corresponding storage needs to be allocated.
+Order and number or storage size of items read must be exactly the same
+as when writing, or else the data will be read incorrectly.
+
+Reading uses the :cpp:func:`utils::sfread <LAMMPS_NS::utils::sfread>`
+utility function to detect read errors and short reads, so that LAMMPS
+can abort if that happens, e.g. when the restart file is corrupted.
+
+.. code-block:: c++
+
+   /* ----------------------------------------------------------------------
+      proc 0 reads from restart file, bcasts
+   ------------------------------------------------------------------------- */
+
+   void PairBornGauss::read_restart(FILE *fp)
+   {
+     read_restart_settings(fp);
+
+     allocate();
+
+     int i, j;
+     int me = comm->me;
+     for (i = 1; i <= atom->ntypes; i++) {
+       for (j = i; j <= atom->ntypes; j++) {
+         if (me == 0) utils::sfread(FLERR, &setflag[i][j], sizeof(int), 1, fp, nullptr, error);
+         MPI_Bcast(&setflag[i][j], 1, MPI_INT, 0, world);
+         if (setflag[i][j]) {
+           if (me == 0) {
+             utils::sfread(FLERR, &biga0[i][j], sizeof(double), 1, fp, nullptr, error);
+             utils::sfread(FLERR, &alpha[i][j], sizeof(double), 1, fp, nullptr, error);
+             utils::sfread(FLERR, &biga1[i][j], sizeof(double), 1, fp, nullptr, error);
+             utils::sfread(FLERR, &beta[i][j], sizeof(double), 1, fp, nullptr, error);
+             utils::sfread(FLERR, &r0[i][j], sizeof(double), 1, fp, nullptr, error);
+             utils::sfread(FLERR, &cut[i][j], sizeof(double), 1, fp, nullptr, error);
+           }
+           MPI_Bcast(&biga0[i][j], 1, MPI_DOUBLE, 0, world);
+           MPI_Bcast(&alpha[i][j], 1, MPI_DOUBLE, 0, world);
+           MPI_Bcast(&biga1[i][j], 1, MPI_DOUBLE, 0, world);
+           MPI_Bcast(&beta[i][j], 1, MPI_DOUBLE, 0, world);
+           MPI_Bcast(&r0[i][j], 1, MPI_DOUBLE, 0, world);
+           MPI_Bcast(&cut[i][j], 1, MPI_DOUBLE, 0, world);
+         }
+       }
+     }
+   }
+
+   /* ----------------------------------------------------------------------
+      proc 0 reads from restart file, bcasts
+   ------------------------------------------------------------------------- */
+
+   void PairBornGauss::read_restart_settings(FILE *fp)
+   {
+     if (comm->me == 0) {
+       utils::sfread(FLERR, &cut_global, sizeof(double), 1, fp, nullptr, error);
+       utils::sfread(FLERR, &offset_flag, sizeof(int), 1, fp, nullptr, error);
+       utils::sfread(FLERR, &mix_flag, sizeof(int), 1, fp, nullptr, error);
+     }
+     MPI_Bcast(&cut_global, 1, MPI_DOUBLE, 0, world);
+     MPI_Bcast(&offset_flag, 1, MPI_INT, 0, world);
+     MPI_Bcast(&mix_flag, 1, MPI_INT, 0, world);
+   }
+
+Writing coefficients to data files
+""""""""""""""""""""""""""""""""""
+
+The ``write_data()`` and ``write_data_all()`` functions are optional and
+write out the current state of the :doc:`pair_coeff
```

Comment:
Also explain this is only relevant for true pairwise styles.  Manybody pair styles don't write to data files.